ArjunaJTA/jdbc/tests/classes/com/hp/mwtests/ts/jdbc/basic/ConnectionImpleTest.java

```diff
@@ -125,13 +124,14 @@ public void beforeCompletion() { }
             public void afterCompletion(int status) {
                 try {
                     connectionToTest.isClosed();
+                    connectionToTest.close();
```

Comment:
But unit test is there to test specific units of code. It doesn't matter if it complies with the spec or not. Integration and functional tests should make sure that the functionality works according to all specs. 
As for moving to the different framework, we should worry about that if the time comes IMO.